elasticsearch/src/main/java/org/hibernate/search/backend/elasticsearch/impl/ElasticsearchHSQueryImpl.java

```diff
@@ -367,6 +353,20 @@ private JsonObject getEffectiveFilter(JsonArray typeFilters) {
 				filters.add( ToElasticsearch.fromLuceneQuery( query ) );
 			}
 
+			// user filter
+			if ( userFilter != null ) {
+				filters.add( ToElasticsearch.fromLuceneFilter( userFilter ) );
+			}
+
+			if ( !filterDefinitions.isEmpty() ) {
+				for ( FullTextFilterImpl fullTextFilter : filterDefinitions.values() ) {
+					JsonObject filter = buildFullTextFilter( fullTextFilter );
+					if ( filter != null ) {
+						filters.add( filter );
+					}
+				}
+			}
+
 			// wrap filters into must if there is more than one
 			return ToElasticsearch.condition( "must", filters );
```

Comment:
in the example I see here:
 - https://www.elastic.co/guide/en/elasticsearch/reference/2.0/query-filter-context.html
the various filters are not connected over a "must" element.

This doesn't seem wrong, but it might be redundant right? Just wondering if you noticed that, or if there is some additional reason.